java_src/foam/dao/history/HistoryDAO.java

```diff
@@ -70,9 +69,8 @@ public FObject put(FObject obj) {
     FObject current = this.find(obj);
 
     // add new history record
-    String objectId = (String) ((PropertyInfo) obj.getClassInfo().getAxiomByName("objectId")).f(obj);
     HistoryRecord historyRecord = new HistoryRecord();
-    historyRecord.setObjectId(objectId);
+    historyRecord.setObjectId(obj.getClassInfo().getId());
```

Comment:
All FObjects stored in a DAO have an id property, either directly or indirectly. If you don't have a property called 'id', then you do ids: [ 'keypart1', ..., 'keypartn' ] and it makes you a psedo-property named 'id' which is an alias to your id(s).